src/ol_infrastructure/components/services/k8s.py

```diff
@@ -0,0 +1,311 @@
+            kubernetes.core.v1.EnvFromSourceArgs(
+                secret_ref=kubernetes.core.v1.SecretEnvSourceArgs(
+                    name=self.ol_app_k8s_config.db_creds_secret_name,
+                ),
+            ),
+            # Redis Configuration
+            kubernetes.core.v1.EnvFromSourceArgs(
+                secret_ref=kubernetes.core.v1.SecretEnvSourceArgs(
+                    name=self.ol_app_k8s_config.redis_creds_secret_name,
+                ),
+            ),
+            # static secrets from secrets-application/secrets
+            kubernetes.core.v1.EnvFromSourceArgs(
+                secret_ref=kubernetes.core.v1.SecretEnvSourceArgs(
+                    name=self.ol_app_k8s_config.static_secrets_name,
+                ),
+            ),
+        ]
+
+        init_containers = [
+            # Run database migrations at startup
+            kubernetes.core.v1.ContainerArgs(
+                name="migrate",
+                image=f"mitodl/unified-application-app-main:{self.ol_app_k8s_config.application_docker_tag}",
+                command=["python3", "manage.py", "migrate", "--noinput"],
+                image_pull_policy="IfNotPresent",
+                env=application_deployment_env_vars,
+                env_from=application_deployment_envfrom,
+            ),
+            kubernetes.core.v1.ContainerArgs(
+                name="collectstatic",
+                image=f"mitodl/unified-application-app-main:{self.ol_app_k8s_config.application_docker_tag}",
+                command=["python3", "manage.py", "collectstatic", "--noinput"],
+                image_pull_policy="IfNotPresent",
+                env=application_deployment_env_vars,
+                env_from=application_deployment_envfrom,
+                volume_mounts=[
+                    kubernetes.core.v1.VolumeMountArgs(
+                        name="staticfiles",
+                        mount_path="/src/staticfiles",
+                    ),
+                ],
+            ),
+        ] + [
+            kubernetes.core.v1.ContainerArgs(
+                name=f"promote-{mit_username}-to-superuser",
+                image=f"mitodl/unified-application-app-main:{self.ol_app_k8s_config.application_docker_tag}",
+                # Jank that forces the promotion to always exit successfully
+                command=["/bin/bash"],
+                args=[
+                    "-c",
+                    f"./manage.py promote_user --promote --superuser '{mit_username}@mit.edu'; exit 0",  # noqa: E501
+                ],
+                image_pull_policy="IfNotPresent",
+                env=application_deployment_env_vars,
+                env_from=application_deployment_envfrom,
+            )
+            for mit_username in DEVOPS_MIT
+        ]
+
+        # Create a deployment resource to manage the application pods
+        application_labels = self.ol_app_k8s_config.k8s_global_labels | {
+            "ol.mit.edu/application": "unified-application",
```

Comment:
We will want this to be the actual application name in order to be able to e.g. differentiate logs in Grafana.